contracts/pangolin-periphery/PangolinBridgeMigrationRouter.sol

```diff
@@ -0,0 +1,358 @@
+import "../pangolin-lib/libraries/TransferHelper.sol";
+import "./interfaces/IBridgeToken.sol";
+import "./libraries/Roles.sol";
+import "./libraries/PangolinLibrary.sol";
+import "hardhat/console.sol";
+
+contract PangolinBridgeMigrationRouter {
+    using SafeMath for uint;
+    using Roles for Roles.Role;
+
+    Roles.Role private adminRole;
+    mapping(address => address) public bridgeMigrator;
+
+    constructor() public {
+        adminRole.add(msg.sender);
+    }
+
+    // safety measure to prevent clear front-running by delayed block
+    modifier ensure(uint deadline) {
+        require(deadline >= block.timestamp, 'PangolinBridgeMigrationRouter: EXPIRED');
+        _;
+    }
+
+    // makes sure the admin is the one calling protected methods
+    modifier onlyAdmin() {
+        require(adminRole.has(msg.sender), 'PangolinBridgeMigrationRouter: Unauthorized');
+        _;
+    }
+
+    /**
+     * @notice Given an address, this address is added to the list of admin.
+     * @dev Any admin can add or remove other admins, careful.
+     * @param account The address of the account.
+     */
+    function addAdmin(address account) external onlyAdmin {
+        adminRole.add(account);
+    }
+
+    /**
+     * @notice Given an address, this address is added to the list of admin.
+     * @dev Any admin can add or remove other admins, careful.
+     * @param account The address of the account.
+     */
+    function removeAdmin(address account) external onlyAdmin {
+        adminRole.remove(account);
+    }
+
+    /**
+     * @notice Given an address, returns whether or not he's already an admin
+     * @param account The address of the account.
+     * @return Whether or not the account address is an admin.
+     */
+    function isAdmin(address account) external view returns(bool) {
+        return adminRole.has(account);
+    }
+
+    /**
+     * @notice Given an token, and it's migrator address, it configures the migrator for later usage
+     * @param token The ERC20 token address that will be migrated using the migrator
+     * @param migrator The WrappedERC20 token address that will be migrate the token
+     */
+    function addMigrator(address token, address migrator) external onlyAdmin {
+        uint256 amount = IBridgeToken(migrator).swapSupply(token);
+        require(
+            amount > 0,
+            "The migrator doesn't have swap supply for this token"
+        );
+        _allowToken(token, migrator);
+        bridgeMigrator[token] = migrator;
+    }
+
+    /**
+     * @notice Internal function to manage approval, allows an ERC20 to be spent to the maximum
+     * @param tokenAddress The ERC20 token address that will be allowed to be used
+     * @param spenderAddress Who's going to spend the ERC20 token
+     */
+    function _allowToken(address tokenAddress, address spenderAddress) internal {
+        IPangolinERC20(tokenAddress).approve(spenderAddress, uint(-1));
```

Comment:
instead of uint(-1),` type(uint256).max` is preferred for readability and removes abiguity

Reply:
I'm going forward with this change, good suggestion!

Reply:
Fixed, resolving